bin/python/ripple/util/Sign.py

```diff
@@ -0,0 +1,139 @@
+    assert len(b) < 192, 'Too long'
+    return chr(len(b)) + b
+
+#-----------------------------------------------------------
+
+def make_seed(urandom=os.urandom):
+    # This is not used.
+    return urandom(16)
+
+def make_ed25519_keypair(urandom=os.urandom):
+    private_key = urandom(32)
+    return private_key, ed25519.publickey(private_key)
+
+def make_ecdsa_keypair():
+    # This is not used.
+    private_key = ecdsa.SigningKey.generate(curve=ecdsa.SECP256k1)
+    # Can't be unit tested easily - need a mock for ecdsa.
+    vk = private_key.get_verifying_key()
+    sig = private_key.sign('message')
+    assert vk.verify(sig, 'message')
+    return private_key, vk
+
+def make_manifest(pk, vpk, seq):
+    return ''.join([
+        SField.sfSequence,
+        to_bytes_bigend(seq, 4),
+        SField.sfPublicKey,
+        prepend_length_byte(pk),
+        SField.sfSigningPubKey,
+        prepend_length_byte(vpk)])
+
+def sign_manifest(manifest, private_key, pk):
+    sig = ed25519.signature('MAN\0' + manifest, private_key, pk)
+    return manifest + SField.sfSignature + prepend_length_byte(sig)
+
+def wrap(s, cols=60):
+    if s:
+        size = max((len(s) + cols - 1) / cols, 1)
+        w = len(s) / size
+        s = '\n'.join(s[i:i + w] for i in range(0, len(s), w))
+    return s
+
+def create_ed_keys(urandom=os.urandom):
+    private_key, pk = make_ed25519_keypair(urandom)
+    apk = ED_BYTE + pk
+    pkh = Base58.encode_version(Base58.VER_NODE_PUBLIC, apk)
+    skh = Base58.encode_version(Base58.VER_NODE_PRIVATE, private_key)
+    v0, apk0 = Base58.decode_version(pkh)
+    assert v0 == Base58.VER_NODE_PUBLIC
+    assert apk0 == apk
+    return pkh, skh
+
+def create():
+    pkh, skh = create_ed_keys()
+    print('[validators]', pkh, '', '[master_secret]', skh, sep='\n')
+
+def check(s):
+    ver, b = Base58.decode_version(s)
+    print('version        = ' + Base58.version_name(ver))
+    print('decoded length = ' + str(len(b)))
+    h = Base58.encode_version(ver, b)
+    assert h == s
+
+def check_validator_public(v, avpk):
+    assert v == Base58.VER_NODE_PUBLIC, 'Not a validator-public.'
+    assert len(avpk) == 33, 'Length of validating key must be 33.'
+    assert ord(avpk[0]) in (2, 3), 'First validating key byte must be 2 or 3.'
+
+def check_master_secret(v, private_key):
+    assert v == Base58.VER_NODE_PRIVATE, 'Version is not private.'
+    assert len(private_key) == 32, 'Length of master secret must be 32.'
+
+def get_signature(seq, vpkh, skh):
+    v, avpk = Base58.decode_version(vpkh)
+    check_validator_public(v, avpk)
+
+    v, private_key = Base58.decode_version(skh)
+    check_master_secret(v, private_key)
+
+    pk = ed25519.publickey(private_key)
+    apk = ED_BYTE + pk
+    m = make_manifest(apk, avpk, seq)
+    m1 = sign_manifest(m, private_key, pk)
+    return base64.b64encode(m1)
+
+def sign(seq, vpkh, skh):
+    print('[validation_manifest]')
+    print(wrap(get_signature(int(seq), vpkh, skh)))
```

Comment:
Its already a constant, the number 60! Seriously though, this is just supposed to be useful interim utility until we implement more robust functionality.